tests/plugins.test.ts

```diff
@@ -473,6 +483,120 @@ test('plugin config order', async () => {
+    expect(pluginConfig.plugin!.capabilities!.scheduled_tasks).toHaveLength(0)
+})
+
+test('plugin with archive loads all capabilities, no random caps', async () => {
+    getPluginRows.mockReturnValueOnce([
+        mockPluginWithArchive(`
+            function processEvent (event, meta) { event.properties={"x": 1}; return event }
+            function randomFunction (event, meta) { return event}
+            function onEvent (event, meta) { return event }
+
+            function runEveryHour(meta) {console.log('1')}
+
+            export const jobs = {
+                x: (event, meta) => console.log(event)
+            }
+        `),
+    ])
+    getPluginConfigRows.mockReturnValueOnce([pluginConfig39])
+    getPluginAttachmentRows.mockReturnValueOnce([pluginAttachment1])
+
+    await setupPlugins(mockServer)
+    const { pluginConfigs } = mockServer
+
+    const pluginConfig = pluginConfigs.get(39)!
+    expect(pluginConfig.plugin!.capabilities!.methods!.sort()).toEqual(['onEvent', 'processEvent', 'processEventBatch'])
+    expect(pluginConfig.plugin!.capabilities!.jobs).toEqual(['x'])
+    expect(pluginConfig.plugin!.capabilities!.scheduled_tasks).toEqual(['runEveryHour'])
+})
+
+test('plugin with source file loads capabilities', async () => {
+    const [plugin, unlink] = mockPluginTempFolder(`
+        function processEvent (event, meta) { event.properties={"x": 1}; return event }
+        function randomFunction (event, meta) { return event}
+        function onEvent (event, meta) { return event }
+    `)
+
+    getPluginRows.mockReturnValueOnce([plugin])
+    getPluginConfigRows.mockReturnValueOnce([pluginConfig39])
+    getPluginAttachmentRows.mockReturnValueOnce([pluginAttachment1])
+
+    await setupPlugins(mockServer)
+    const { pluginConfigs } = mockServer
+
+    const pluginConfig = pluginConfigs.get(39)!
+    expect(pluginConfig.plugin!.capabilities!.methods!.sort()).toEqual(['onEvent', 'processEvent', 'processEventBatch'])
+    expect(pluginConfig.plugin!.capabilities!.jobs).toEqual([])
+    expect(pluginConfig.plugin!.capabilities!.scheduled_tasks).toEqual([])
+
+    unlink()
+})
+
+test('plugin with source code loads capabilities', async () => {
+    const source_code = `
+        function processEvent (event, meta) { event.properties={"x": 1}; return event }
+        function randomFunction (event, meta) { return event}
+        function onSnapshot (event, meta) { return event }
+    `
+    getPluginRows.mockReturnValueOnce([mockPluginSourceCode(source_code)])
+
+    getPluginConfigRows.mockReturnValueOnce([pluginConfig39])
+    getPluginAttachmentRows.mockReturnValueOnce([pluginAttachment1])
+
+    await setupPlugins(mockServer)
+    const { pluginConfigs } = mockServer
+
+    const pluginConfig = pluginConfigs.get(39)!
+
+    expect(pluginConfig.plugin!.capabilities!.methods!.sort()).toEqual([
+        'onSnapshot',
+        'processEvent',
+        'processEventBatch',
```

Comment:
It seems a bit quirky that `processEventBatch` is listed as a capability when it's only an autogenerated version based on actually provided `processEvent`, but that seems OK